### PR TITLE
Cucumber instance variable refactor

### PR DESCRIPTION
**Pull request by jvanderhoof**
### Desired Outcome

The goal of this PR is twofold:

1. Provide clarity of information shared between Scenario Steps. This is accomplished via a Scenario Context, which holds any information needed in other Scenario Steps. This approach provides an alternative to instance variables, which lack context.
2. Create a clear inventory if information required to run a particular Feature. This is accomplished with the ability to define  context variables and their sources (environment variable and/or a default value). This is combined with the ability to define Conjur variables and values and tabular form. The intent is to provide a more declarative approach to setting up Features and Scenarios.

### Implemented Changes

*Describe how the desired outcome above has been achieved with this PR. In
particular, consider:*

- _What's changed? Why were these changes made?_
- _How should the reviewer approach this PR, especially if manual tests are required?_
- _Are there relevant screenshots you can add to the PR description?_

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

CyberArk internal issue ID: [insert issue ID]

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

**Review by adamouamani — COMMENTED** [review]
Looks good @jvanderhoof - I like the data tables approach you created as it makes it very clear what you are doing and data values being used in test.  I also like how you created a `scenario_context` and leverage this to share information for multiple tests!!  Do we document this anywhere - would be nice to formalize this more as a best practice.  I can help with that - would be good to have a place to reference.

File: cucumber/authenticators_jwt/features/authn_jwt_check_standard_claims.feature [hunk 1]
```diff
@@ -68,8 +71,11 @@ Feature: JWT Authenticator - Check registered claim
       role: !group conjur/authn-jwt/raw/hosts
       member: !host myapp
     """
-    And I am the super-user
-    And I successfully set authn-jwt "token-app-property" variable to value "host"
+    And I set the following conjur variables:
+      | variable_id                                   | default_value |
```
Comment (adamouamani) [inline cucumber/authenticators_jwt/features/authn_jwt_check_standard_claims.feature:75]: @jvanderhoof I like this pattern - it makes it easier to follow and understand and it simplifies the flow

File: cucumber/authenticators_oidc/features/authn_oidc.feature [hunk 2]
```diff
@@ -35,8 +44,10 @@ Feature: OIDC Authenticator - Hosts can authenticate with OIDC authenticator
       role: !group conjur/authn-oidc/keycloak/users
       member: !user alice
     """
-    And I am the super-user
-    And I successfully set OIDC variables
+    And I set the following conjur variables:
```
Comment (adamouamani) [inline cucumber/authenticators_oidc/features/authn_oidc.feature:47]: @jvanderhoof - Do we need to add variable or note to feature description that this is run as admin user?

File: cucumber/authenticators_oidc/features/authn_oidc_bad_policy.feature [hunk 3]
```diff
@@ -33,8 +43,9 @@ Feature: OIDC Authenticator - Bad authenticator configuration leads to an error
       role: !group conjur/authn-oidc/keycloak/users
       member: !user alice
     """
-    And I am the super-user
-    And I successfully set provider-uri variable
+    And I set the following conjur variables:
+      | variable_id                             | context_variable  |
+      | conjur/authn-oidc/keycloak/provider-uri | oidc_provider_uri |
     And I fetch an ID Token for username "alice" and password "alice"
     And I save my place in the log file
```
Comment (adamouamani) [inline cucumber/authenticators_oidc/features/authn_oidc_bad_policy.feature:50]: @jvanderhoof - with the goal of being more declarative - I'm wondering if this is required as a separate step?  could this be included as part of another step?

File: cucumber/authenticators_oidc/features/authn_oidc.feature (hunk 2 above)
Comment (jvanderhoof) [reply cucumber/authenticators_oidc/features/authn_oidc.feature:47]: @adamouamani, ironically, the line `I am the super-user` didn't actually do anything. Policy and variables are loaded using the `admin` user.  There are a couple of cases (the scenarios which continue to use that line), where we need the line (typically when accessing something related to the file system).

Comment (adamouamani) [reply cucumber/authenticators_oidc/features/authn_oidc.feature:47]: @jvanderhoof as we were discussing, these steps should be more declarative and reusable.  The line about `I am the super-user` seems like something for background describing persona.  I'm also wondering how we can use your pattern of data tables and scenario context to make steps definitions more reusable and flexible

**Review by adamouamani — APPROVED** [review]
Reviewing these changes has been great and we have identified some great patterns to apply going forward.  This is awesome!!

**Pull request merged**